nanome/api/plugin.py

```diff
@@ -55,25 +69,99 @@ def run(self, host = "config", port = "config", key = "config"):
         :type host: str
         :type port: int
         """
-        if (host == "config"):
-            self.__host = config.fetch("host")
-        else:
-            self.__host = host
-        if (port == "config"):
-            self.__port = config.fetch("port")
-        else:
-            self.__port = port
-        if (key == "config"):
-            self.__key = config.fetch("key")
-        else:
-            self.__key = key
-        self.__parse_args()
+        default_host = config.fetch('host') if host == 'config' else host
+        default_port = config.fetch('port') if port == 'config' else port
+        default_key = config.fetch('key') if key == 'config' else key
+
+        # Parse command line args and set internal variables.
+        parser = self.create_parser()
+        args = parser.parse_args()
+
+        self.host = args.host or default_host
+        self.port = args.port or default_port
+        self.key = args.keyfile or default_key
+        self.has_autoreload = args.auto_reload
+
+        self.verbose = args.verbose
+        Logs._set_verbose(self.verbose)
+
+        if args.ignore:
+            to_ignore = args.ignore.split(",")
+            self.to_ignore.extend(to_ignore)
+
+        # Name can be set during the class instantiation without cli flag.
+        if args.name and not self.name:
+            self.name = ' '.join(args.name)
+
         Logs.debug("Start plugin")
-        if self.__has_autoreload:
+        if self.has_autoreload:
             self.__autoreload()
         else:
             self.__run()
 
+    @property
```

Comment:
I don't think these getters and setters are useful, especially since it's useless to set values of the Plugin object after calling run()